app/assets/javascripts/admin/addon/components/admin-branding-fonts-form.gjs

```diff
@@ -0,0 +1,286 @@
+          count,
+        },
+      });
+      await this.#save(data);
+    } else {
+      await this.#save(data);
+    }
+  }
+
+  @action
+  async #save(data) {
+    try {
+      await ajax("/admin/config/branding/fonts.json", {
+        type: "PUT",
+        data: {
+          base_font: data.base_font,
+          heading_font: data.heading_font,
+          default_text_size: data.default_text_size,
+          update_existing_users: this.updateExistingUsers,
+        },
+      });
+      this.toasts.success({
+        duration: 3000,
+        data: {
+          message: i18n("admin.config.branding.fonts.form.saved"),
+        },
+      });
+      this.siteSettingChangeTracker.refreshPage();
+    } catch (err) {
+      this.toasts.error({
+        duration: 3000,
+        data: {
+          message: err.jqXHR.responseJSON.errors[0],
+        },
+      });
+    }
+  }
+
+  @cached
+  get formData() {
+    return {
+      base_font: this.siteSettings.base_font,
+      heading_font: this.siteSettings.heading_font,
+      default_text_size: this.siteSettings.default_text_size,
+    };
+  }
+
+  <template>
+    <Form
+      @onSubmit={{this.update}}
+      @data={{this.formData}}
+      class="admin-fonts-form"
+      as |form transientData|
+    >
+      <form.Field
+        @name="base_font"
+        @title={{i18n "admin.config.branding.fonts.form.base_font.title"}}
+        @validation="required"
+        @format="full"
+        as |field|
+      >
+        <field.Custom>
+          {{#each FONTS as |font|}}
+            <DButton
+              @action={{fn this.setButtonValue field.set font}}
+              class={{concatClass
+                "admin-fonts-form__button-option font btn-flat"
+                (decamelize (underscore font))
+                (if
+                  (eq transientData.base_font (decamelize (underscore font)))
+                  "active"
+                )
+              }}
+            >{{font}}</DButton>
+          {{/each}}
+          {{#if this.moreBaseFonts}}
+            {{#each MORE_FONTS as |font|}}
+              <DButton
+                @action={{fn this.setButtonValue field.set font}}
+                class={{concatClass
+                  "admin-fonts-form__button-option font btn-flat"
+                  (decamelize (underscore font))
+                  (if
+                    (eq transientData.base_font (decamelize (underscore font)))
+                    "active"
+                  )
+                }}
+              >{{font}}</DButton>
+            {{/each}}
+          {{/if}}
+          <DButton
+            @action={{this.toggleMoreBaseFonts}}
+            class="admin-fonts-form__more font"
+          >
+            {{#if this.moreBaseFonts}}
+              {{i18n "admin.config.branding.fonts.form.less_fonts"}}
+            {{else}}
+              {{i18n "admin.config.branding.fonts.form.more_fonts"}}
+            {{/if}}
+          </DButton>
```

Comment:
This component is being repeated twice as well, I think there is value in extracting this into another component.

Reply:
Maybe consider moving the `form.field` into a component instead since it is pretty much a duplicate between the base fonts and heading fonts.